docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,439 @@
+core libraries (ie. the verification library and the implementation of core
+traits by the Tendermint types) will be re-usable by IBC.
+
+## Decision
+
+Below is a schematic of the components of a lite node:
+
+![Lite Client Diagram](assets/adr-002-image.png)
+
+We take up the components of the diagram in term.
+
+### State
+
+The lite node state contains the following:
+
+- current height (H)
+- last header (H-1)
+- current validators (H)
+
+It also includes some configuration, which contains:
+
+- trusting period
+- initial list of full nodes
+- method (sequential or skipping)
+- trust level (if method==skipping)
+
+The node is initialied with a trusted header for some height H-1
+(call this header[H-1]), and a validator set for height H (call this vals[H]).
+
+The configuration contains an initial list of peers.
+For the sake of simplicity, one of the peers is selected as the "primary", while the
+rest are considered "backups". Most of the data is downloaded from the primary,
+and double checked against the backups.
+
+The state is considered "expired" if the difference between the current time and
+the time from the trusted header is greater than a configurable "trusting
+period". If at any point the state is expired, the node should log an error and
+exit - it's needs to be manually reset.
+
+
+### Manager
+
+For lack of a better name. The Manager co-ordiantes the syncing and is the
+highest level component. We consider two approaches to syncing the lite node: sequential and skipping.
+
+#### Sequential Sync
+
+Inital state: 
+
+    - time T
+    - height H 
+    - header[H-1]
+    - vals[H]
+
+Here we describe the happy path:
+
+1) Request header[H], commit[H], and vals[H+1] from the primary, and check that they are well formed and from the correct height
+2) Pass header[H], commit[H], vals[H], and vals[H+1] to the verification library, which
+
+  will:
+    - check that vals[H] and vals[H+1] are correctly reflected in the header
+    - check that the commit is for the header
+    - check that +2/3 of the validators correctly signed the header hash
+3) Request header[H] from each of the backups and check that they match header[H] received from the primary
+4) Update the state with header[H] and vals[H+1], and increment H
+5) return to (1)
+
+If (1) or (2) fails, mark the primary as bad and select a new peer to be the
+primary.
+
+If (3) returns a conflicting header, verify the header by requesting the
+corresponding commit and running the verification of (2). If the verification
+passes, there is a fork, and evidence should be published so the validators get
+slashed. We leave the mechanics of evidence to a future document. For now, the
+lite client will just log an error and exit. If the verification fails, it
+means the backup that provided the conflict is bad and should be removed.
+
+#### Skipping Sync
+
+Skipping sync is essentially the same as sequential, except for a few points:
```

Comment:
Is there a spec / security analysis I can read for this? It seems unsafe to me without full node updates or validity proofs. (SNARKs)

Reply:
spec is here, should have linked to it properly: https://github.com/tendermint/tendermint/blob/master/docs/spec/consensus/light-client.md